File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/response/type/ConnectNodeSetImpulse.kt
package gaia.sdk.response.type

import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.response.intf.*
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator
import gaia.sdk.request.enumeration.*

/**
* Impulse which indicates the result setting a node connection
*/
data class ConnectNodeSetImpulse @JsonCreator constructor(
    @JsonProperty("id") val id:Uuid? = null, 
    /**
    * edges removed before setting the new edge
    */
    @JsonProperty("removedEdges") val removedEdges:List<EdgeKeyOne>? = null, 
    /**
    * edge created by this request
    */
    @JsonProperty("newEdge") val newEdge:Edge? = null
)

File: gaia-sdk-python/gaia_sdk/graphql/response/type/ConnectNodeSetImpulse.py

from typing import List

from gaia_sdk.graphql.response.type.Edge import Edge
from gaia_sdk.graphql.response.type.EdgeKeyOne import EdgeKeyOne

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class ConnectNodeSetImpulse:
    """
    Impulse which indicates the result setting a node connection
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    edges removed before setting the new edge
    """
    @property
    def removed_edges(self) -> List[EdgeKeyOne]:
        return list(map(lambda x: EdgeKeyOne(x), self.dictionary.get("removedEdges")))
    """
    edge created by this request
    """
    @property
    def new_edge(self) -> Edge:
        return Edge(self.dictionary.get("newEdge"))
